acceptance/framework/environment/environment.go

```diff
@@ -59,6 +61,20 @@ func NewKubernetesEnvironmentFromConfig(config *config.TestConfig) *KubernetesEn
 		kenv.contexts[SecondaryContextName] = NewContext(config.SecondaryKubeNamespace, config.SecondaryKubeconfig, config.SecondaryKubeContext)
 	}
 
+	// Optionally, deploy apps into a separate namespace.
+	// Maybe not the right place for this?
```

Comment:
If this will always be the `consul` namespace then wouldn't we want to deploy the helm chart as well as applications into the restricted one? Or do you think we'll be deploying the helm chart into one and the applications into another one. Also, if it's always the same, we could also consider hard-coding it and setting it whenever `enable-openshift` flag is set.

Reply:
Oh also, maybe of the `*_namespaces*` tests already deploy apps in their own namespaces so that would probably interfere with these settings as they will get overridden by the specific tests. For example: https://github.com/hashicorp/consul-k8s/blob/c790951382235e9a5f10365d1b46fba16c5f2e5a/acceptance/tests/connect/connect_inject_namespaces_test.go#L90-L99

Reply:
>  My thought was those tests would ultimately use the specified app + app secondary namespaces (if specified) instead of the hard-coded ones.

What happens if a test wants a 3rd app installed in another namespace? 4th? 

We are already seeing a similar thing show up with contexts in the partition tests where we now need 3 clusters and multiple services/namespaces setup...

Reply:
That's true. In any case, I've removed the app namespace flags, so that tests can choose what namespaces they use, so resolving this thread.